pkg/manifests/manifests.go

```diff
@@ -1425,6 +1429,10 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, trustedCABundleCM *v1.Config
 		p.Spec.Retention = f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.Retention
 	}
 
+	if !zeroRetentionSizeConfig(f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.RetentionSize) {
```

Comment:
I wonder should the previous check `if f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.Retention != ""` also check for zero values? 

If yes then perhaps we can rename `zeroRetentionSizeConfig` to `isUnsetRetentionConfig` or something similar

Reply:
Trying to cover the scenario If the retention value is "", 0 or "0", it means that the value isn't defined, so `isUnsetRetentionConfig` name makes sense